src/core/optimization_container.jl

```diff
@@ -657,13 +656,18 @@ function get_initial_conditions(
 end
 
 function get_initial_conditions(optimization_container::OptimizationContainer, key::ICKey)
-    return get_initial_conditions(optimization_container.initial_conditions, key)
+    initial_conditions = get(optimization_container.initial_conditions, key, nothing)
+    if initial_conditions === nothing
+        throw(IS.InvalidValue("initial conditions are not stored for $(key)"))
```

Comment:
In similar cases with variables and parameters we print a sorted list of keys that are present. That can help debug the problem.